### PR TITLE
Fix cards not deleting properly.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Fixes cards not deleting properly.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51879

### CODE REVIEW

**Review by sbishel — CHANGES_REQUESTED** [review]
A couple minor things, clean up.

File: server/app/blocks.go [hunk 1]
```diff
@@ -303,6 +303,14 @@ func (a *App) DeleteBlock(blockID string, modifiedBy string) error {
...
 }
 
+func retriveFileIDFromBlockFieldStorage(id string) string {
+	parts := strings.Split(id, ".")
+	if len(parts) < 1 {
+		return ""
+	}
+	return parts[0][1:]
+}
+
```
[the remark this answers is not in the document]
Comment (crspeller) [reply server/app/blocks.go:313]: Yup. All that logic got moved to the store along with a version of this function.

File: server/services/store/sqlstore/blocks.go [hunk 2]
```diff
@@ -415,6 +423,27 @@ func (s *SQLStore) deleteBlockAndChildren(db sq.BaseRunner, blockID string, modi
...
+	// fileId and attachmentId shoudn't exist at the same time
+	fileID := ""
+	fileIdWithExtention, fileIDExists := block.Fields["fileId"]
+	if fileIDExists {
+		fileID = retriveFileIDFromBlockFieldStorage(fileIdWithExtention.(string))
+	}
+	attachmentIDWithExtention, attachmentIDExists := block.Fields["attachmentId"]
+	if attachmentIDExists {
+		fileID = retriveFileIDFromBlockFieldStorage(attachmentIDWithExtention.(string))
+	}
```
[the remark this answers is not in the document]
Comment (crspeller) [reply server/services/store/sqlstore/blocks.go:435]: Yup I guess we can.

**Review by sbishel — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

**Comment by sbishel** [comment]
Will need to cp -> release-7.10, release-7.9, release-7.8.

**Review by esarafianou — APPROVED** [review]
LGTM!

**Pull request merged**